src/containers/GroupAdmin/components/MemberListItem.tsx

```diff
@@ -0,0 +1,108 @@
+import { Box, Button, Collapse, Divider, ListItem, ListItemText, makeStyles } from '@material-ui/core';
+import Paper from 'components/layout/Paper';
+import { useState } from 'react';
+import ExpandMoreIcon from '@material-ui/icons/ExpandMoreRounded';
+import ExpandLessIcon from '@material-ui/icons/ExpandLessRounded';
+import { User } from 'types/Types';
+import { getUserClass, getUserStudyShort } from 'utils';
+import HighlightOffIcon from '@material-ui/icons/HighlightOff';
+import StarIcon from '@material-ui/icons/Star';
+import classNames from 'classnames';
+import { useDeleteMembership, useUpdateMembership } from 'api/hooks/Membership';
+import useSnackbar from 'api/hooks/Snackbar';
+import { MembershipType } from 'types/Enums';
+import Dialog from 'components/layout/Dialog';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    marginBottom: theme.spacing(2),
+  },
+  button: {
+    textTransform: 'none',
+  },
+  gutterBottom: {
+    marginBottom: theme.spacing(2),
+  },
+}));
+
+type MemberListItemProps = {
+  user: User;
+  slug: string;
+};
+
+const MemberListItem = ({ slug, user }: MemberListItemProps) => {
+  const classes = useStyles();
+  const [expanded, setExpanded] = useState(false);
+  const deleteMembership = useDeleteMembership(slug, user.user_id);
+  const updateMembership = useUpdateMembership(slug, user.user_id);
+  const showSnackbar = useSnackbar();
+  const [confirmDeleteModal, setConfirmDeleteModal] = useState(false);
+  const [confirmSwitchLeader, setConfirmSwitchLeader] = useState(false);
+  const handleClose = () => {
+    setConfirmDeleteModal(false);
+    setConfirmSwitchLeader(false);
+  };
+
+  const removeMemberFromGroup = () => {
+    deleteMembership.mutate(null, {
+      onSuccess: (data) => {
+        showSnackbar(data.detail, 'success');
+      },
+      onError: (e) => {
+        showSnackbar(e.detail, 'error');
+      },
+    });
+  };
+
+  const promoteUserToLeader = () => {
+    updateMembership.mutate(MembershipType.LEADER, {
+      onSuccess: () => {
+        showSnackbar('Brukeren ble promotert til leder', 'success');
+      },
+      onError: (e) => {
+        showSnackbar(e.detail, 'error');
+      },
+    });
+  };
+  return (
+    <Paper className={classes.root} noPadding>
+      {confirmDeleteModal && (
+        <Dialog
+          onClose={handleClose}
+          onConfirm={removeMemberFromGroup}
+          open={confirmDeleteModal}
+          titleText={`Fjern ${user.first_name} ${user.last_name} fra gruppen?`}></Dialog>
+      )}
+      {confirmSwitchLeader && (
+        <Dialog
+          onClose={handleClose}
+          onConfirm={promoteUserToLeader}
+          open={confirmSwitchLeader}
+          titleText={`Promoter ${user.first_name} ${user.last_name} til leder?`}></Dialog>
+      )}
```

Comment:
Bør ikke være nødvendig å legge Dialogene inne i conditional rendering
```suggestion
        <Dialog
          onClose={handleClose}
          onConfirm={removeMemberFromGroup}
          open={confirmDeleteModal}
          titleText={`Fjern ${user.first_name} ${user.last_name} fra gruppen?`} />
        <Dialog
          onClose={handleClose}
          onConfirm={promoteUserToLeader}
          open={confirmSwitchLeader}
          titleText={`Promoter ${user.first_name} ${user.last_name} til leder?`} />
```